guava-tests/test/com/google/common/reflect/ClassPathTest.java

```diff
@@ -153,6 +160,41 @@ public void testScan_classPathCycle() throws IOException {
     }
   }
 
+  public void testScanDirectory_symlinkCycle() throws IOException {
+    ClassLoader loader = ClassPathTest.class.getClassLoader();
+    // directory with a cycle,
+    // /root
+    //    /left
+    //       /[sibling -> right]
+    //    /right
+    //       /[sibling -> left]
+    Path root = createTempDirectory("ClassPathTest");
+    try {
+      Path left = createDirectory(root.resolve("left"));
+      createFile(left.resolve("some.txt"));
+
+      Path right = createDirectory(root.resolve("right"));
+      createFile(right.resolve("another.txt"));
+
+      createSymbolicLink(left.resolve("sibling"), right);
+      createSymbolicLink(right.resolve("sibling"), left);
+
+      ClassPath.DefaultScanner scanner = new ClassPath.DefaultScanner();
+      scanner.scan(root.toFile(), loader);
+
+      assertEquals(ImmutableSet.of(
```

Comment:
Lines 185-192 don't look like they're exactly in Google Java Style to me. If you haven't done so already, I suggest running [google-java-format](https://github.com/google/google-java-format) on this file.